csharp/ql/lib/semmle/code/csharp/controlflow/Guards.qll

```diff
@@ -287,6 +287,18 @@ private module LogicInput implements GuardsImpl::LogicInputSig {
     g2 = g1.(NullCoalescingExpr).getAnOperand() and
     v1.isNullValue() and
     v2 = v1
+    or
+    exists(Assertion assert, AssertMethod target, int i |
+      assert.getAssertMethod() = target and
+      g1 = assert and
+      v1.getDualValue().isThrowsException() and
+      g2 = assert.getExpr(i)
+    |
+      target.(BooleanAssertMethod).getAnAssertionIndex(v2.asBooleanValue()) = i
+      or
+      target.(NullnessAssertMethod).getAnAssertionIndex(any(boolean isNull | v2.isNullness(isNull))) =
+        i
+    )
```

Comment:
For my understanding: Would the dual also be correct
```ql
    exists(Assertion assert, AssertMethod target, int i |
      assert.getAssertMethod() = target and
      g1 = assert and
      v1.isThrowsException() and
      g2 = assert.getExpr(i)
    |
      target.(BooleanAssertMethod).getAnAssertionIndex(v2.asBooleanValue().getDualValue()) = i
      or
      target.(NullnessAssertMethod).getAnAssertionIndex(any(boolean isNull | v2.isNullness(isNull.booleanNot()))) =
        i
    )
```
I realize that adding it would not make a difference, since nothing can be controlled by the exit-edge out of the assertion, but if it had instead been an exception edge, that could in principle happen.